neuralop/layers/discrete_continuous_convolution.py

```diff
@@ -0,0 +1,648 @@
+    """
+    Discrete-continuous convolutions (DISCO) on arbitrary 2d grids as implemented for [1]. To evaluate continuous convolutions on a computer, they can be evaluated semi-discretely, where the translation operation is performed continuously, and the quadrature/projection is performed discretely on a grid [2]. They are the main building blocks for local Neural Operators [1]. Forward call expects an input of shape batch_size x in_channels x n_in.
+
+    Parameters
+    ----------
+    in_channels: int
+        input channels to DISCO convolution
+    out_channels: int
+        output channels of DISCO convolution
+    grid_in: torch.Tensor or str
+        input grid in the form of a point cloud. Can also pass a string to generate a regular (tensor) grid.
+    grid_out: torch.Tensor or str
+        output grid in the form of a point cloud. Can also pass a string to generate a regular (tensor) grid.
+    kernel_shape: Union[int, List[int]]
+        kernel shape. Expects either a signle integer for isotropic kernels or two integers for anisotropic kernels
+    n_in: Tuple[int], optional
+        number of input points
+    n_out: Tuple[int], optional
+        number of output points
+    quad_weights: torch.Tensor, optional
+        quadrature weights on the input grid
+    periodic: bool, optional
+        whether the domain is periodic
+    groups: int, optional
+        number of groups in the convolution
+    bias: bool, optional
+        whether to use a bias
+    radius_cutoff: float, optional
+        cutoff radius for the kernel
+
+    References
+    ----------
+    .. [1] Liu-Schiaffini M., Berner J., Bonev B., Kurth T., Azizzadenesheli K., Anandkumar A.; Neural Operators with Localized Integral and Differential Kernels;  arxiv:2402.16845
+    .. [2] Ocampo J., Price M.A. , McEwen J.D.; Scalable and equivariant spherical CNNs by discrete-continuous (DISCO) convolutions, ICLR (2023), arXiv:2209.13603
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        grid_in: torch.Tensor,
+        grid_out: torch.Tensor,
+        kernel_shape: Union[int, List[int]],
+        n_in: Optional[Tuple[int]] = None,
+        n_out: Optional[Tuple[int]] = None,
+        quad_weights: Optional[torch.Tensor] = None,
+        periodic: Optional[bool] = False,
+        groups: Optional[int] = 1,
+        bias: Optional[bool] = True,
+        radius_cutoff: Optional[float] = None,
+    ):
+        super().__init__(in_channels, out_channels, kernel_shape, groups, bias)
+
+        # the instantiator supports convenience constructors for the input and output grids
+        if isinstance(grid_in, torch.Tensor):
+            assert isinstance(quad_weights, torch.Tensor)
+            assert not periodic
+        elif isinstance(grid_in, str):
+            assert n_in is not None
+            assert len(n_in) == 2
+            x, wx = _precompute_grid(n_in[0], grid=grid_in, periodic=periodic)
```

Comment:
given that _precompute_grid is in another lib, can we name x, wx with self explanatory names and/or add a short comment explaining they're the quadrature weights?